### ISSUE TITLE
SkipGram.java 186 throw NullPointerException(batches.get()  is null)

### ISSUE THREAD

**Issue by chenying99** [deeplearning4j/deeplearning4j#2415]
hi   this is my test:

int batchSize = 1000;
		int iterations = 3;
		int layerSize = 150;

		LOGGER.info("Build model....");
		Word2Vec vec = new Word2Vec.Builder().batchSize(batchSize) // # words
																	// per
																	// minibatch.
				.minWordFrequency(5) //
				.useAdaGrad(false) //
				.layerSize(layerSize) // word feature vector size
				.iterations(iterations) // # iterations to train
				.learningRate(0.025) //
				.minLearningRate(1e-3) // learning rate decays wrt # words.
										// floor learning
				.negativeSample(10) // sample size 10 words
				.iterate(iterator) //
				.tokenizerFactory(tokenizerFactory).build();

		LOGGER.info("Fitting Word2Vec model....");
		vec.fit();

throw exception :

14:22:07.428 [main] DEBUG o.d.m.w.wordstore.VocabConstructor - Vocab size after truncation: [421506],  NumWords: [80955965], sequences parsed: [5322815], counter: [83608778]
14:22:08.497 [main] INFO  o.d.m.w.wordstore.VocabConstructor - Sequences checked: [5322815], Current vocabulary size: [421506]; Sequences/sec: [30521.60];
14:22:09.131 [main] INFO  o.d.m.e.inmemory.InMemoryLookupTable - Initializing syn1...
14:22:09.483 [main] INFO  o.d.m.s.SequenceVectors - Building learning algorithms:
14:22:09.483 [main] INFO  o.d.m.s.SequenceVectors -           building ElementsLearningAlgorithm: [SkipGram]
14:22:09.487 [main] INFO  o.d.m.s.SequenceVectors - Starting learning process...
Exception in thread "VectorCalculationsThread 6" java.lang.RuntimeException: java.lang.NullPointerException
	at org.deeplearning4j.models.sequencevectors.SequenceVectors$VectorCalculationsThread.run(SequenceVectors.java:1100)
Caused by: java.lang.NullPointerException
	at org.deeplearning4j.models.embeddings.learning.impl.elements.SkipGram.learnSequence(SkipGram.java:186)
	at org.deeplearning4j.models.sequencevectors.SequenceVectors.trainSequence(SequenceVectors.java:303)
	at org.deeplearning4j.models.sequencevectors.SequenceVectors$VectorCalculationsThread.run(SequenceVectors.java:1075)

SkipGram.java:186

if (batches.get().size() >= configuration.getBatchSize()) {
			Nd4j.getExecutioner().exec(batches.get());
			batches.get().clear();
		} 

because batches.get()  is null

**Comment by raver119** [comment]
This issue is already fixed on current master.
